server/task/src/test/java/org/apache/james/task/eventsourcing/TerminationSubscriberContract.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * *************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ***************************************************************/
+
+package org.apache.james.task.eventsourcing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.apache.james.eventsourcing.Event;
+import org.apache.james.eventsourcing.EventId;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskId;
+
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public interface TerminationSubscriberContract {
+    TerminationSubscriber subscriber();
+
+    @Test
+    default void handlingCompletedShouldBeListed() {
+        TerminationSubscriber subscriber = subscriber();
+        TaskEvent event = new Completed(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42), Task.Result.COMPLETED);
+
+        subscriber.handle(event);
+
+        assertThat(subscriber.listenEvents().limit(1)).containsOnly(event);
```

Comment:
```
assertThat(Stream.of(x))
     .first()
     .isEqualTo(y);
```

Seems more straightforward....

Reply:
sure, but that's not the intent

Reply:
What's your intent, and how my proposal fails to express it then?

I fear I need additional information here...

Reply:
The intent is: given a `Stream` of acceptable `Event`, it gives a `Stream` of these `Event`s.

The fact that this `Stream` is a singleton is accidental.

Reply:
Semantically these 2 proposal are the same.

From what you say <I understand you want to assert:

```
assertThat(Stream.of(x)).containsOnly(x);
```

(dropping the limit)

Reply:
No really.

Anyway, it's an infinite blocking `Stream`, this code does not work.